bam_to_mate_hist.py

```diff
@@ -89,14 +90,16 @@ def parse_bam_file(bamfile, num_reads, count_diff_refname_stub=False):
         zero_dists = 0
         mapq0_reads = 0
         num = 0
-        dists = np.empty([num_reads, 1], dtype=int)
+        large_insert_possible = 0
+        large_insert_actual = 0
+        dists = {}
```

Comment:
could do 
`dists = defaultdict(lambda: defaultdict(int))`
to make it simpler to handle unset keys below. but not a huge deal.